% Template for PLoS
% Version 3.1 February 2015
%
% To compile to pdf, run:
% latex plos.template
% bibtex plos.template
% latex plos.template
% latex plos.template
% dvipdf plos.template
%
% % % % % % % % % % % % % % % % % % % % % %
%
% -- IMPORTANT NOTE
%
% This template contains comments intended 
% to minimize problems and delays during our production 
% process. Please follow the template instructions
% whenever possible.
%
% % % % % % % % % % % % % % % % % % % % % % % 
%
% Once your paper is accepted for publication, 
% PLEASE REMOVE ALL TRACKED CHANGES in this file and leave only
% the final text of your manuscript.
%
% There are no restrictions on package use within the LaTeX files except that 
% no packages listed in the template may be deleted.
%
% Please do not include colors or graphics in the text.
%
% Please do not create a heading level below \subsection. For 3rd level headings, use \paragraph{}.
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% -- FIGURES AND TABLES
%
% Please include tables/figure captions directly after the paragraph where they are first cited in the text.
%
% DO NOT INCLUDE GRAPHICS IN YOUR MANUSCRIPT
% - Figures should be uploaded separately from your manuscript file. 
% - Figures generated using LaTeX should be extracted and removed from the PDF before submission. 
% - Figures containing multiple panels/subfigures must be combined into one image file before submission.
% For figure citations, please use "Fig." instead of "Figure".
% See http://www.plosone.org/static/figureGuidelines for PLOS figure guidelines.
%
% Tables should be cell-based and may not contain:
% - tabs/spacing/line breaks within cells to alter layout or alignment
% - vertically-merged cells (no tabular environments within tabular environments, do not use \multirow)
% - colors, shading, or graphic objects
% See http://www.plosone.org/static/figureGuidelines#tables for table guidelines.
%
% For tables that exceed the width of the text column, use the adjustwidth environment as illustrated in the example table in text below.
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% -- EQUATIONS, MATH SYMBOLS, SUBSCRIPTS, AND SUPERSCRIPTS
%
% IMPORTANT
% Below are a few tips to help format your equations and other special characters according to our specifications. For more tips to help reduce the possibility of formatting errors during conversion, please see our LaTeX guidelines at http://www.plosone.org/static/latexGuidelines
%
% Please be sure to include all portions of an equation in the math environment.
%
% Do not include text that is not math in the math environment. For example, CO2 will be CO\textsubscript{2}.
%
% Please add line breaks to long display equations when possible in order to fit size of the column. 
%
% For inline equations, please do not include punctuation (commas, etc) within the math environment unless this is part of the equation.
%
% % % % % % % % % % % % % % % % % % % % % % % % 
%
% Please contact latex@plos.org with any questions.
%
% % % % % % % % % % % % % % % % % % % % % % % %

\documentclass[10pt,letterpaper]{article}
\usepackage[top=0.85in,left=2.75in,footskip=0.75in]{geometry}

% Use adjustwidth environment to exceed column width (see example table in text)
\usepackage{changepage}

% Use Unicode characters when possible
\usepackage[utf8]{inputenc}

% textcomp package and marvosym package for additional characters
\usepackage{textcomp,marvosym}

% fixltx2e package for \textsubscript
\usepackage{fixltx2e}

% amsmath and amssymb packages, useful for mathematical formulas and symbols
\usepackage{amsmath,amssymb}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

% Use nameref to cite supporting information files (see Supporting Information section for more info)
\usepackage{nameref,hyperref}

% line numbers
\usepackage[right]{lineno}

% ligatures disabled
\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% rotating package for sideways tables
\usepackage{rotating}

% Remove comment for double spacing
%\usepackage{setspace} 
%\doublespacing

% Text layout
\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.25in 
\textheight 8.75in

% Bold the 'Figure #' in the caption and separate it from the title/caption with a period
% Captions will be left justified
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off]{caption}

% Use the PLoS provided BiBTeX style
\bibliographystyle{plos2015}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

% Leave date blank
\date{}

% Header and Footer with logo
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
\lhead{\includegraphics[width=2.0in]{PLOS-submission.eps}}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}
\lfoot{\sf PLOS}

%% Include all macros below

\newcommand{\lorem}{{\bf LOREM}}
\newcommand{\ipsum}{{\bf IPSUM}}

%% END MACROS SECTION


\begin{document}
\vspace*{0.35in}

% Title must be 250 characters or less.
% Please capitalize all terms in the title except conjunctions, prepositions, and articles.
\begin{flushleft}
{\Large
\textbf\newline{Multi-omic vaginal microbiome profiling reveals bacterial vaginosis subtypes}
}
\newline
% Insert author names, affiliations and corresponding author email (do not include titles, positions, or degrees).
\\
Jean M. Macklaim\textsuperscript{1,2},
Amy McMillan\textsuperscript{1,3},
Jo-Anne Hammond\textsuperscript{4},
Michael John\textsuperscript{5},
Mark Sumarah\textsuperscript{6},
Jonathan Swann\textsuperscript{7},
Gregor Reid\textsuperscript{1,3,8},
Gregory B. Gloor\textsuperscript{1,2,*},
with the VOGUE Research Group\textsuperscript{\textpilcrow}
\\
\bigskip
\bf{1} Canadian Centre for Human Microbiome and Probiotic Research, Lawson Health Research Institute, The University of Western Ontario, London, Ontario, Canada
\\
\bf{2} Department of Biochemistry, The University of Western Ontario, London, Ontario, Canada
\\
\bf{3} Department of Microbiology and Immunology, The University of Western Ontario, London, Ontario, Canada
\\
\bf{4} Department of Family Medicine, The University of Western Ontario, London, N6A 5C1, Canada
\\
\bf{5} Department of Pathology and Laboratory Medicine, The University of Western Ontario, London, N6A 5C1, Canada
\\
\bf{6} Agriculture and Agri-food Canada, London, Ontario, Canada
\\
\bf{7} Division of Computational and Systems Medicine, Department of Surgery and Cancer, Imperial College London, UK.
\\
\bf{8} Department of Surgery, The University of Western Ontario, London, Ontario, Canada
\\
\bigskip

% Insert additional author notes using the symbols described below. Insert symbol callouts after author names as necessary.
% 
% Remove or comment out the author notes below if they aren't used.
%
% Primary Equal Contribution Note
%\Yinyang These authors contributed equally to this work.

% Additional Equal Contribution Note
% Also use this double-dagger symbol for special authorship notes, such as senior authorship.
%\ddag Corresponding Author

% Current address notes
%\textcurrency a Insert current address of first author with an address update
% \textcurrency b Insert current address of second author with an address update
% \textcurrency c Insert current address of third author with an address update

% Deceased author note
%\dag Deceased

% Group/Consortium Author Note
\textpilcrow Membership list can be found in the Acknowledgments section.

% Use the asterisk to denote corresponding authorship and provide email address in note below.
* ggloor@uwo.ca

\end{flushleft}
% Please keep the abstract below 300 words
\section*{Abstract}
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur eget porta erat. Morbi consectetur est vel gravida pretium. Suspendisse ut dui eu ante cursus gravida non sed sem. Nullam sapien tellus, commodo id velit id, eleifend volutpat quam. Phasellus mauris velit, dapibus finibus elementum vel, pulvinar non tellus. Nunc pellentesque pretium diam, quis maximus dolor faucibus id. Nunc convallis sodales ante, ut ullamcorper est egestas vitae. Nam sit amet enim ultrices, ultrices elit pulvinar, volutpat risus.


% Please keep the Author Summary between 150 and 200 words
% Use first person. PLOS ONE authors please skip this step. 
% Author Summary not valid for PLOS ONE submissions.   
%\section*{Author Summary}
%The meta-transcriptome of twenty-two human vaginal samples was characterized at 
\linenumbers

\section*{Introduction}

One major challenge of microbial investigation is to determine the  role of the microbiome in the environment	. We can divide microbiome analyses into categories of increasing resolution that correspond to the question answered: `who is there' can be addressed by 16S rRNA gene sequencing (or other target gene sequencing method); `what is the capacity of the microbiome' can be addressed by full metageomic sequencing or by an extrapolation method;  `what have they produced' can be addressed by small molecule profiling (metabolomics); and `what are they doing now' can be addressed by transcriptome profiling. The vast majority of microbiome analyses are designed to address the first question, and more recently methods have been developed to estimate the second question from target gene sequencing profiles. It is widely recognized that these two approaches are largely descriptive. However, the ability to address the latter two questions is less widespread because of their high complexity in both sample collection and in the analysis.  

Amy - an introductory paragraph on metabolome

There are several barriers to collecting and analyzing whole microbial community transcriptomes (meta-transcriptomes). First, RNA is a much less stable molecule than DNA, with a significant fraction of microbial genes having a half-life under two minutes, and almost all having a half-life less than 10 minutes \cite{mRNA:2002, mRNA:2006}. Thus, careful sample collection and storage is extremely important to preserve the integrity of the sample. Second, unlike DNA abundance, RNA abundances are not necessarily tied to organism abundance because RNA production is induced or repressed in response to environmental cues. Thus, any method that discriminates based on the abundance of an RNA molecule will be at risk of confounding the effect of organism abundance with the abundance of the RNA species. The wide dynamic range of RNA abundances also makes assembly of the sequencing output problematic since many assembly algorithms assume a somewhat constant read coverage. Third, many RNA-seq experiments have a very high background of reads that do not map or assemble into sensible contigs: the origin of these phantom reads is unknown, but they greatly reduce the amount of information available on a meta-transcriptome at all sequencing depths. These constraints mean that many meta-transcriptome experiments yield less information than expected, and in some cases. 

A final issue is that high-throughput sequencing data are compositional: that is, they are constrained to an arbitrary constant sum \cite{Friedman:2012, fernandes:2013, Lovell:2015}. Compositional data have the property of linked correlations. This means that as one feature (gene or organism) becomes more abundant, one or more others \emph{must} become relatively less abundant. This can be seen easily in the trivial example of a two part component $x=[1,9]$ that originally has 10 parts and the two parts are partitioned such that $x_1$=10\% and $x_2$=90\%. . If, for example, the first part increases, $x_1 \to 9$,  and the second part remains unchanged then the two parts are now partitioned such that   $x_1$=50\% and $x_2$=50\%. Note that it appears that $x_2$ has become less abundant, when in fact it has remained unchanged. With a large number of such samples, we would conclude that the two abundances are negatively correlated, when in fact they are not correlated.

The problem of compositional data analysis is general no matter the number of features (genes or organisms). The linked correlation problem is the root cause of spurious correlation: where changing the membership of a composition by adding or removing one or more features, changes the correlation structure in unpredictable ways \cite{Aitchison:1986}. In many cases, the sign of  correlation between  features can change from strongly positive to strongly negative \cite{Friedman:2012, Lovell:2015}. This means that the observed correlations depend on group membership and not on any intrinsic correlation. There is not a current consensus on how to deal with this problem, and  groups have attempted to minimize the problem \cite{Friedman:2012, Kurtz:2015} or to approach the problem analytically \cite{Lovell:2015}.  Spurious correlation is very problematic because many multivariate tools such as ordination, principle component analysis and clustering are dependent upon it. 

The spurious correlation problem was recognized very early by Pearson \cite{Pearson:1896}, yet a general solution was not elaborated until Aitchison suggested a log-ratio approach \cite{Aitchison:1986}. The approach advocated by Aitchison was to convert the data into ratios between the features and then to take the logarithm to make the data symmetrical such that the choice of numerator or denominator would change only the sign, and not the  value of the result. Since then, much work has been done to place this approach onto a firm theoretical and practical foundation (e.g. \cite{pawlowsky2015modeling,MPE2011,Egozcue:2003,martin:2003}), although these methods have only recently begun to be noticed and used by biologists  \cite{Friedman:2012, fernandes:2013, macklaim:2013, ancom:2015, Lovell:2015, Kurtz:2015}. 

%http://statisticalhorizons.com/zero-inflated-models: zero inflated models assume that the data come from a mixture where some of the samples are really 0 and some are not. i.e. the data are generated by a mxiture of models where one models the likelihood of 0 and not 0, and the other models the count likelihood (which can be 0). These are point estimates and do not convert to probabilities.
A final complication is that the measurement error of compositional data is different than that expected for non-compositional data being non-linearly related to the read depth \cite{fernandes:2013}. The measurement error issue has led to a number of distinct approaches that aim to estimate the underlying distribution of these data. Zero-inflated negative binomial methods are currently popular since they appear to be able to provide good point estimates for the distributions of these highly sparse datasets. However, estimates derived from these approaches are highly parameterized and as such may be overfitting the data.

Rather than using point estimates, these datasets can be analyzed in a Bayesian manner, where the observations (count per feature) serve as the prior for the estimation of the posterior distribution of probabilities \cite{fernandes:2013,fernandes:2014, gloorAJS:2016}. Using this approach a prior of zero can be converted to a posterior distribution of non-zero, probabilities that can be small or large depending on the sequencing depth and the number of features in the sample. We have shown that the posterior closely models the actual measurement error of compositional data and that that approach \cite{fernandes:2013, gloorAJS:2016}. One added benefit of this approach is that the need for sequencing-depth normalization is removed when the Bayesian estimates are used in concert with the log-ratio transformation \cite{fernandes:2013, Lovell:2015}. This is because the distribution of the posterior becomes broader when sequencing depth is low, reflecting the increased precision in this case. Thus, it is observed that low sequencing depth simply reduces power,  as expected when less data is collected. 



It has been argued that that microbial communities are best though of as ensembles of genes since the genetic makeup of individual bacterial genomes is rather fluid \cite{boon:2014}. 

%\begin{equation}\label{eq:schemeP} 
%D_{coll} = \frac{D_f+\frac{[S]^2}{K_D S_T} D_S} {1+\frac{[S]^2}{K_D S_T}}, 
%D_{sm} = \frac{D_f+ \frac{[S]}{K_D} D_S}{1+\frac{[S]}{K_D}},
%\end{equation}

% You may title this section "Methods" or "Models". 
% "Models" is not a valid title for PLoS ONE authors. However, PLoS ONE
% authors may use "Analysis" 
\section*{Materials and Methods}
\subsection*{Ethics.}
\subsection*{Collection.}
\subsection*{RNA isolation and sequencing.}
\subsection*{Data analysis.} 

\paragraph{Read Mapping.} XX genomes comprising the set of species observed in the human vagina from 16S rRNA gene sequencing (REFS), culture studies (REFS) and additional genomes from organisms suspected to occur were downloaded from Genbank on ????. Open reading frames from these genomes were clustered using ?? with the following criteria (length, PID, etc) and a representative sequence was chosen to be the centroid using the following rule: ???. Centroid sequences were annotated by BLAST to SEED and KEGG databases and the best hit supplying the annotation. The taxonomy of the centroid sequences was taken to represent the taxonomy of the cluster. We will refer to the centroid sequence as the `transcript'. Supplementary Table S1 contains the list of accession numbers. Supplementary Table S2 contains the set of centroid sequences and their inferred annotation. 

Reads in fastq format were mapped to the library of centroid sequences using bowtie2 \cite{bowtie2} and unmapped reads were assembled with ???. The assembled fragments were annotated as above, and added to the master table of counts per sequence feature in each sample. In the end there were an average of XXX million reads mapped to each sample, and Table S3 contains the pertinent information. Reads were further grouped by KEGG function or SEED level 4 subsystem to produce two tables of functions, these will be referred to as `functions'.

\paragraph{Statistical Model for RNA-seq.} The table of counts mapped to reference sequences, and aggregated to either the SEED subsystem or KEGG KO level, appears on the surface to be a table of counts. Common practice is to normalize the  reads per feature such that the `sequencing effort' is a constant or each sample \cite{Anders:2013aa}. However, we  have found this approach  fails in meta-RNA sequencing because of the interplay between the organism and transcript abundance \cite{fernandes:2013,fernandes:2014,Lovell:2015}. Thus, it much more informative to model the observed count for a feature in a sample as a distribution of probabilities that the count was observed given the total number of sequence reads obtained per sample \cite{fernandes:2013}. Such a model falls into the count compositional data analysis paradigm \cite{Aitchison:1986} where only the relative differences between abundances in a sample are informative \cite{pawlowsky2015modeling,pawlowsky2011compositional}. Upon reflection, this approach makes intrinsic sense because the number of reads obtained in any high throughput sequencing experiment is constrained by the capacity of the instrument. We know intuitively that the same library will give a more accurate depiction of reality with the $\sim 200$ million reads obtained from a HiSeq lane than with the $\sim 20$ million reads obtained from a MiSeq.

Compositional data has substantially different properties than do unconstrained data. The most important being that compositional data exhibits `spurious correlation' between features because the possible values for  the features are constrained by the constant sum \cite{pawlowsky2015modeling,pawlowsky2011compositional}. Correlations are also unreliable because they change in unpredictable ways when a subset of the parts are examined. Thus, traditional correlation metrics are unsuitable when examining these data \cite{Lovell:2015}. Note that  all RNA-seq datasets are subcompositions unless ribosomal RNA, tRNA, etc are not depleted and are included in the analysis \cite{pawlowsky2015modeling,pawlowsky2011compositional}. 

Some of these problems can be ameliorated by log-ratio transformations such as the centered log-ratio transformation (clr). Given a vector of numbers that contains D features, $x=[x_1,x_2, ... x_D]$,   $x$ can be converted to a vector of clr values, $z$, as follows: 
\begin{equation}\label{eq:clr}
\mathrm{clr}(x)=\left(\ln \frac{x_1}{g_x}, \ln\frac{x_2}{g_x}, \dots, \ln\frac{x_D}{g_x}\right)=\left(z_1,z_2, \dots, z_n \right) \,.
\end{equation}

Where $g_x$ is the geometric mean of vector $x$.

One complication is that the geometric mean cannot be computed when the $x$ contains a value of 0. Thus two approaches were taken. First, when a point estimate was required  the `count zero multiplicative' zero replacement method from the zCompositions R package \cite{PalareaAlbaladejo201585} was used to adjust 0 values for the likelihood that the 0 represents a non-detect event in the sample. Second, when conducting quantitative analyses, the joint probability distributions for all features in a sample were generated by Dirichlet multinomial sampling within the ALDEx2 Bioconductor R package  \cite{fernandes:2013} using a uniform adjustment of Jeffrey's Prior. This returns a distribution of possible values for the probability of  the feature given the feature count and the total count for the sample \cite{fernandes:2013, gloorAJS:2016}. We have shown previously that the relative error in these probability distributions is largest for features with 0 counts and smallest for those with the largest number of counts in both real and simulated data \cite{fernandes:2013}. The distributions were subsequently transformed using the centre log-ratio transformation  prior to further analysis. 

Exploratory data analysis was performed as point estimates with compositional biplots \cite{aitchison2002biplots} following clr transformation of the data. These show both the distances between samples and the variances of the transcripts.

The expected value of $\phi$ \cite{Lovell:2015} from the clr distribution was used to determine compositionally-linked transcripts or functions, since it has been shown that traditional correlation metrics give unpredictable results\cite{pawlowsky2015modeling,pawlowsky2011compositional}. The expected value of Kendall's Tau (b) was used when reporting the correlation between transcript abundance and metabolite abundance since it is not expected that the metabolome and meta-transcriptome tables share even passingly similar units or scaling. We used an effect size cutoff of 2 when evaluating differential abundance between conditions, a cutoff that was used in our previous investigations \cite{macklaim:2013}. 

\subsection*{Metabolome determination.}

\paragraph{Sample preparation GC-MS.} Samples were collected from the mid-vaginal wall using the Cytobrush vaginal brush and sterile forceps and stored at -80$^\circ$C until analysis. Vaginal brushes were pre-cut into 1.5 mL tubes and weighed prior to and after sample collection to determine the mass of vaginal fluid collected. After thawing, brushes were eluted in methanol-water (1:1) to a final concentration of 0.05 g/mL. This corresponded to a volume of 200-1500 $\mu$L, depending on the mass of vaginal fluid collected. Four control swabs were included which consisted of blank swabs eluted in 200, 400, 800, or 1500  $\mu$L of methanol-water. Samples and controls were vortexed for 10 sec to extract metabolites, centrifuged for 5 min at 10 000 rpm, vortexed again for 10 sec after which time the brushes were removed from tubes. Samples were centrifuged a final time to pellet cells and 150  $\mu$L of supernatant transferred to GC-MS vials. Remaining supernatant was transferred to a new 1.5 mL tube, frozen at -80$^\circ$C and shipped to the University of Reading for NMR analysis. Next, 25  $\mu$L of 0.2 mg/ml ribitol standard was added to each GC-MS vial. Samples were then dried to completeness using a SpeedVac. After drying 100  $\mu$L of 2\% methoxyamine-HCl in pyridine (MOX) was added to each sample for derivatization and samples were incubated at 500C for 90 min. 100  $\mu$L N-Methyl-N-(trimethylsilyl) trifluoroacetamide (MSTFA) was then added to each vial and incubated at 500C for 30 min. After derivitization, an equal aliquot of each sample was combined to make the quality control (QC). Samples were then transferred to micro inserts before analysis by GC-MS (Agilent 7890A GC, 5975 inert MSD with triple axis detector). One  $\mu$L of sample was injected using pulsed splitless mode into a 30 m DB5-MS column with 10 m duraguard, diameter 0.35mm, thickness 0.25  $\mu$m (JNW Scientific). Helium was used as the carrier gas at a constant flow rate of 1 mL/min. Oven temperature was held at 70$^\circ$C for 5 min then increased at a rate of 5$^\circ$C/min to 300$^\circ$C and held for 10 min. Solvent delay was set to 13 min to avoid solvent and a large lactate peak, and total run time was 61 min. Masses between 25 m/z and 600 m/z were selected by the detector. All samples were run in random order and a the QC was run multiple times throughout the run to ensure machine consistency. 

\paragraph{Data analysis GC-MS.} Chromatogram files were de-convoluted and converted to ELU format using the AMDIS Mass Spectrometry software \cite{stein:1999} with the sensitivity set to medium. Chromatograms were then aligned and integrated using Spectconnect \cite{stycz:2007} software with the Support Threshold set to low. All metabolites found in the blank swab, or believed to have originated from derivatization reagents were removed from analysis at this time. After removal of swab metabolites, the IS matrix from Spectconnect was transformed using the additive log ratio transformation (alr) and ribitol as a normalizing agent, $log2(x) / log2(\mathrm{ribitol})$. Zeros were replaced with two thirds the minimum detected value on a per metabolite basis prior to transformation. All further metabolite analysis was performed using these alr transformed values. 
	
A total of 90 metabolites were detected by GC-MS. Upon inspection it was determine that 50 of these metabolites were either redundant, background noise or present in controls and were removed from analysis. For redundant peaks, the sum of each peak was combined resulting in a single value for each metabolite. Metabolites were initially identified by comparison to the NIST 11 standard reference database (http://www.nist.gov/srd/nist1a.cfm). Identities of metabolites of interest were then confirmed by authentic standards if available. 

Independent Wilcox tests with a Benjamini-Hochberg correction to account for multiple testing were used to determine metabolites that differed significantly between healthy and BV (p $<$ 0.10.). Groups were defined as healthy or BV using a percentage \emph{Lactobacillus} cutoff of 75\%

\paragraph{Sample preparation NMR.} Samples were dried to completeness and resuspended in phosphate buffer saline (pH 7) containing sodium azide and the internal standard Trimethylsilyl propanoic acid (TSP). Samples were run on a Bruker 700 MHz cryoprobe spectrometer. A standard one-dimensional NMR spectrum was acquired for each sample with water peak suppression using a standard pulse sequence (recycle delay (RD)-90 -t1-90 -tm-90 -acquire free induction decay(FID)). The RD was set at 2 s and the mixing time (tm)100 ms. The pulse length was approximately 12   $\mu$s and t1 was set to 3  $\mu$s. For each sample, 8 dummy scans were followed by 64 scans and collected in 64 K data points using a spectral width of 20 ppm and an acquisition time per scan of 2.73 s.


\paragraph{Data analysis NMR.}  Spectra were processed according to the methods of Swann et al 2011 \cite{Swann:2011} with the following modifications. 1H NMR spectra were manually corrected for phase and baseline distortions and then referenced to the TSP resonanceat $\delta$0.0. Spectra were digitized using an in-house MATLAB(version R2009b, The Mathworks, Inc.; Natwick, MA) script. To prevent baseline effects that arise from imperfect water saturation the region containing the water resonance was excised. An in-house peak alignment algorithm was then performed on each spectrum in MATLAB to adjust for shifts in peak position due to small pH differences between samples and then each spectrum was normalized using a sum normalization approach.  Principal components analysis (PCA) using pareto scaling was applied in SIMCA (Umetrics, Umea). Orthogonal projection to latent structure discriminant analysis (OPLS-DA) models were constructed using unit variance scaling to aid the interpretation of the model and distinguish the metabolites that differed between the groups. Here, 1H NMR spectroscopic data were used as the descriptor matrix and class information (N or BV) as the response variable. The contribution of each variable (metabolite) to sample classification was visualized by back-scaling transformation, generating a correlation coefficient plot. These coefficient plots are colored according to the significance of correlation to “class” (e.g. N or BV), with red indicating high significance and blue indicating low significance. The direction and degree of the signals relate to covariation of the metabolites with the classes in the model. For all models, one orthogonal component was used to remove systematic variation unrelated to class. Predictive performance was assessed using the Q2$^\wedge$Y parameter. 

\paragraph{Reproducibility.} All R code needed to reproduce this analysis from the count tables can be accessed at: github...


% Results and Discussion can be combined.
\section*{Results and Discussion}

High throughput sequencing experiments generate datasets where the total number of reads per sample are irrelevant, thus these data are compositional and contain only relative information about abundances \cite{fernandes:2014}. Such data can be examined in a rigorous manner by examining the variation in ratios between all pairs of transcripts \cite{Aitchison:1986, pawlowsky2015modeling,pawlowsky2011compositional}.  The first step  centre log-ratio transformation, which reduces each count value to a ratio of that count to the geometric mean count within each sample. The logarithm of the ratio makes any change in relative abundance symmetrical. The clr transformation is functionally equivalent to a matrix of all vs all ratios, and  compensates for differences in read abundance, thus eliminates the need for count normalization \cite{Lovell:2015}.  

%Two caveats need to be expressed at the outset. First, the clr transformation cannot be performed in the presence of transcripts with 0 counts since it involves taking a logarithm. Initially all transcripts that have 0 reads in all samples, so-called `essential zeros' are removed.  We take two approaches for the remaining 0 value. When a point estimate of 0 is required, we use the count zero multiplicative method \cite{PalareaAlbaladejo201585} that replaces each 0 value using a Bayesian estimate of the most likely value of 0 in each sample. This approach is used to adjust the data for the Singular Value Decomposition that is required for Principle Component Analysis (PCA). When a more robust estimate is required, we generate a Bayesian estimate of the range of values for each transcript (including 0 count transcripts) that are consistent with the original dataset \cite{fernandes:2013}. All subsequent analyses are conducted on the resulting distribution of values, and the final reported value is the expected value of the p-value, $\phi$ \cite{Lovell:2015}, or effect-size metric. 

\subsection{Exploratory analysis}
We sequenced 27 vaginal  mRNA samples enriched for microbial mRNA on the Illumina HiSeq platform at The Centre for Applied Genomics in Toronto, and for this analysis we also included in the RNA-seq results of four samples sequenced using the ABI SoLiD platform and reported previously \cite{macklaim:2013}. One purpose of this work was to develop a robust approach for the exploration of RNA-seq datasets using compositional data, or CoDa, approach.  We and others have shown that clr-transforming the data constitutes the first step towards a CoDa analysis approach that is generally useful  to characterize HTS \cite{Friedman:2012, fernandes:2013,fernandes:2014,Lovell:2015}. All data analysis is thus done in a compositional analysis framework with clr-transformed data as outlined in the Materials and Methods.

The workhorse tool for CoDa is the compositional biplot, which summarizes in one plot the relationships between the samples and the contributions of the variables to those relationships \cite{aitchison2002biplots}. A `form' compositional biplot of the dataset at the level of individual refuses (corresponding to genes grouped by X percent identity, see Materials and Methods) is shown in Fig.~\ref{F1:refseq_biplot}refseq. This representation best preserves the relationships between the samples, and places the variables in relation to their contribution to sample location. Although a better representation of the variation in the refseqes is obtained with a `variance' biplot, for the purposes of illustration, in this plot, the distance of a transcript from the origin is  related to the standard deviation of the clr value of the transcript in the samples. Refseq location is \emph{not} directly related to the absolute abundance of the transcript. The direction of the transcript relative to the samples shows the sample group with the greatest abundance of the transcript. Our ability to interpret the distance and direction information for both the samples and the refseqs is only as good as the projection of the PCR, which is determined by the proportion of variance explained on the plot. The biplot in Figure \ref{F1:refseq_biplot} explains 40.6\% of the variance on PC1 and 12.1\% of the variance on PC2. We conclude that this representation is a good one since these two principle components explain over 52\% of the variance in such a large dataset.

There are several observations. First, we can see that the samples partition along PC1 into several groups, with healthy on the left and BV on the right, and that several of the samples partition strongly on PC2. The major taxonomic groups to which the refseqs map is shown by color at the level of species for \emph{Lactobacillus} and at the level of genus for the remainder. It is obvious that the distinction between samples on  PC1 is driven by differential occurrence of \emph{Lactobacillus} species on the left and non-lactobacillus anaerobes on the right. Note that the major lactobacillus groups separate much more strongly on PC2 than do the taxa associated with BV. This could indicate that healthy microbiotas colonized by a near monoculture of one or the other lactobacillus species have distinct ways of being healthy, or it could be an artefact of the non-overlapping gene content of these organisms. The partitioning of different lactobacillus dominated healthy microbiota types is well documented in the literature \cite{Ravel:2010, Hummelen:2010,mcmillan:2015}   and we did not examine this further. 

Interestingly, transcripts annotated as belonging to the ubiquitous\emph{Lactobacillus iners} are near the middle of PC1, indicating that transcripts associated with this organism contribute little to the health-BV separation.  Within BV, we observe that \emph{Megasphaera} and \emph{Prevotella} species form two distinct foci suggesting the presence of distinctive species or strains of these organisms in BV. Interestingly, for \emph{Megasphaera} one of these foci is very close to the origin on PC1, suggesting that this strain or species is contributing little to the overall BV phenotype. Finally, several de-novo assembled transcripts appear to be major contributors to the BV phenotype.

Not surprisingly, the reference sequence-based biplot shows that taxonomic abundance is the major driver of the variation of transcript abundance between samples. Therefore, examining the difference between groups at the level of individual refseqs would provide little more information than could be obtained by knowing the genome of the organisms occurring in each sample. Thus, we were interested to determine if the different states had different underlying functions regardless of their taxonomic composition. 

Refseqs were grouped by SEED subsystem 4 function \cite{Aziz:2008} and Fig. \ref{F1:refseq_biplot}SEED shows the result.  Here we can see that the SEED functions partition more strongly along the major axis of variance with 52.2\% of the variance explained on PC1, and 8.5\% on PC2 when aggregated by SEED subsystem 4.  Thus the first two components explain at least 60\% of the variance. Interestingly, we observe that both the healthy and BV groups appear to partition into two subgroups

We examined the reason for this apparent partitioning using the bar plots and SEED subways4 functional expression profiles in Figure~\ref{F2:barplot}.  Here, we can see that many samples group strongly by similar functional expression profiles. The first group is the H1 group and is composed or largely \emph{L. crispatus} by both the 16S rRNA gene sequencing and the mRNA fraction mapping. This group thus corresponds to the community state type 1 suggested by Ravel and coworkers \cite{Ravel:2010}. The second group, H2, is composed largely of a mixed set of \emph{Lactobacillus} species, often dominated by \emph{L. iners} in total abundance, but with a substantial gene expression contribution from \emph{L. jensenii}, \emph{L. gasseri} or unknown \emph{Lactobacillus} sp. This group could not be neatly put into a community state type. Interestingly we observed that the BV group partitioned into two groups. The BV1 group contained a substantial amount of the unclassified BVAB1 organism by 16S rRNA gene sequencing and had a large gene expression contribution from \emph{Sneathia} sp, and also contained the largest contribution of expression from de-novo assembled contigs. These contigs were assigned to the BVAB1 group color in the figure. The final group, BV2, had only a small amount of BVAB1, and a generally larger amount of \emph{Atopobium} sp. by 16S rRNA gene profiling, and a very small or absent contribution to gene expression by \emph{Sneathia} and BVAB1. 

We found that several samples did not fit neatly into these groups, and had long branches connecting them to all other samples. These samples were found to have very atypical community profiles. For example, samples 013B was composed largely by \emph{Atopobium} sp. by 16S rRNA gene profiling, and by \emph{Megasphaera} by gene expression contribution. Sample 015B was composed of \emph{Bifidobacterium} sp. and \emph{Streptococcus} by 16S rRNA gene profiling, and had a significant contribution from the latter to its gene expression profile. Thus, these two samples exhibited very atypical profiles.   We also excluded four samples from analysis because they had long branches with their nearest neighbour or nearest clearly defined group in the clustered heat map, or because they contained substantial expression of both lactobacillus genes and of non-lactobacillus genes,  or because they contained a substantial fraction of non-lactobacillus organisms by 16S rRNA gene sequencing yet did not cluster with either BV group. These samples include sample 31S which was embedded within the BV group with a long branch, and samples 001A, 003A, 008B. Finally, sample 019A was excluded even though it contained almost exclusively \emph{L. iners} by both 16S rRNA gene profiling and by gene expression composition, yet it did not fit into either of the H groups. Interestingly, this sample was classed as clinically BV by Nugent scoring. We noted that the majority  of these samples were found to be located very near the centre of PC1 on either the refseq or SEED biplots (31S, 003A, 015B, 013B) the suggesting that they were placed inaccurately because of differences in gene expression, or were very far from all other samples (001A, 008B). All further analysis was done using the samples in the H1, H2, BV1 and BV2 groups.  

%%%%
% this is figure of biplots_filter.pdf
%%%%
\begin{figure}[h]
\caption{{\bf Exploratory analysis of the reference sequence transcripts.}
Compositional biplots of the reference sequence transcripts filtered to include transcripts present at an average count of more than 2 reads per sample. Compositional biplots are Principle Component (PC) Plots generated from the singular value decomposition of the centre log-ratio transformed dataset \cite{aitchison2002biplots}. The bottom and left axes show the unit scaled measures for PC1 and 2, and the top and right axes show unit scaled variances for the transcripts. These plots thus show the relationship between sample distance along with the contribution of the transcripts to that distance. In this plot component 1 and 2 explain over 58\% of the variance in the dataset, which is exceptional for a dataset of this complexity. Sample names are shown in black text. The location of each transcript is shown as a coloured point, with the colors corresponding to the taxonomic assignment of the centroid sequence. The second panel shows a blow-up of the area of the biplot on the right side. Taxa are labeled with the  color corresponding to the points and the following legend: Lc-\emph{L. crispatus},  Li-\emph{L. iners},  Lje-\emph{L. jensenii}, Ljg-\emph{L. johnsonii} or \emph{L. gasseri}, Gv-\emph{G. vaginalis}, Me-\emph{Megasphaera} sp., Pr-\emph{Prevotella} sp., AS- assembled transcripts. }
\label{F1:refseq_biplot}
\end{figure}

\subsection{Exploring differential abundance}


. We observe that in both cases the  samples group into the same four groups as observed for the non-aggregated data. This indicates that the observed splits are robust to aggregation of the data, and are not simply driven by changes in taxonomic abundance. Samples composing the healthy group on the left side of the biplot are associated with a relatively small set of functions that are proportionally increased in these samples, and the samples composing the BV group are associated with a large set of functions that are proportionally increased in the BV samples.  The large number of functions with small variance indicates the presence of a core set of functions that are required regardless of condition. The health and BV samples also partition along PC2 similarly to the partitioning seen in Panel A. 

We note that that the majority of functions are at or near the origin with the greatest density being near -5 on PC1. A limitation of the a centered log-ratio approach, or indeed of any ratio-based approach is that the geometric mean of a sample depends on the density of 0 values in the sample.  Samples will have a high density of 0 values in the group that has fewer expressed genes. The H1 and H2 group samples, composed of \emph{Lactobacillus} species will have a much smaller set of functions than will the BV1 and BV2 groups that comprise a mixed bag of organisms, and which generally include at least some functions found only in \emph{Lactobacillus}. Thus the geometric mean for the H groups will be closer to 0 than will the geometric mean for the BV group samples, and the resulting ratio values will be higher. 


The distinctions between the different \emph{Megasphaera} and \emph{Prevotella} types can be seen more clearly in Supplementary Figure 1.


 or by KEGG function \cite{Okuda:2008} 
 
We next observe that there are potential groups of healthy and of BV samples that separate along PC2. The first healthy group, H1, is composed of samples 30S, 4S, 1B, 9B, and 10B. The second healthy group, H2, is composed of samples 2B, 4B, 0B, 1A and 6B. The first BV group, BV1, is composed of samples 16B, 27S, 14B, 13A, 8A, 17B, 18B, 12B, and the second, BV2, is composed of samples 9A, 6A, 10A and 12A. The reason for these groupings is obvious when we examine the taxonomic origin of each transcript.  Here we must recall that there is \emph{no information} regarding absolute abundance for a transcript and that two transcripts with widely varying absolute abundances but similar ratio abundances will have similar locations on the biplot. Inspection of the location of the transcripts in the biplot indicates that they are largely separable by taxonomy. Transcripts mapping to \emph{Lactobacillus crispatus}  furthest to the left and near the bottom correspond to the H1 group. Transcripts derived to the remaining lactobacilli are more closely associated with the H2 group. Such a split has been observed before, and group H1 corresponds to vaginal community state type 1 \cite{Ravel:2010}. The samples in H2 are much less homogenous than those in H1, and correspond  to a mixture of community state types two through 4. 

The BV1 group is associated with an increased ratio abundance of \emph{Leptotrichia} sp., one  \emph{Megasphaera} and one \emph{Prevotella} species and the unmapped assembled transcripts. The BV2 group is more associated with \emph{L. iners}, \emph{Gardnerella vaginalis} and the other strain or species of \emph{Prevotella} sp. and \emph{Megasphaera} sp. 


Traditional statistical approaches, while widely used, are inappropriate when examining compositional  abundances using the $\phi$ metric that assesses the stability of the abundance ratios across samples for all possible pairs of transcripts \cite{Lovell:2015}. Samples that have stable ratios are said to be compositionally associated and have a value of $phi$ near 0. 

As with any compositional data analysis approach, values of 0 are not compatible with the transformation required. In the original paper, the authors removed from analysis all genes that had a 0 count in any sample and then calculated the clr and $\phi$ as point estimates. We took a different approach and estimated the distribution of values that 0 could reasonably assume and then clr-transformed the data using the aldex.clr function from the ALDEx2 Bioconductor package \cite{fernandes:2013, fernandes:2014}. The $\phi$ measure was then calculated for each independent element of the distribution and the expected value of $\phi$ was reported. This approach allowed us to determine reasonable estimates for compositional association values even for functions with a dichotomous distribution, while still maintaining a low false positive background. 

Figure \ref{F3:phi_biplot} shows a graphical depiction of $\phi$ groupings determined for the SEED subsystem 4 functional grouping and the KEGG functional grouping. In both approaches, there were four major clusters that partitioned on PC1, and Supplementary Tables 3 and 4 contain the clusters of SEED subsystem 4 and KO annotations. The first group, composed of XX members was the small group of functions that were much more relatively abundant in the healthy than in the BV group. Inspection of these functions shows that they are belong to ???. These same functions were generally observed to be differentially abundant in our previous analysis (REF). The second group which is nearer the origin was also relatively more abundant in health than in BV, and were found to correspond largely to replication and gene expression housekeeping genes. This confirms our previous observation that these abundant housekeeping genes are slightly increased relative tot he remainder in the healthy microbiome. The third major group had a much greater relative abundance in BV than in health. These functions are largely functions that are absent in lactobacilli and correspond to functions involved in the greater biosynthetic and respiratory capacities of the majority of the BV-associated organisms (REF). In addition to the three major groups, there were a number of smaller clusters. These were generally found to correspond to functions in the same pathway or in many cases to co-expressed subunits.  A full list of functions and their associated group is found in tabular format in the supplement. 


\begin{figure}[h]
\caption{{\bf Compositional association of microbial functions.}
Panel A shows a compositional biplot of the SEED subsystem 4 aggregated data with different groups of compositionally associated transcripts coloured by group membership. Panel B shows the same for data aggregated and analyzed by KEGG annotation. In both cases we determined the expected value of $\phi$ with a cutoff of 0.03. }
\label{F3:phi_biplot}
\end{figure}


\begin{figure}[h]
\caption{{\bf Correlation between metabolite abundance and microbial function.}
 Correlations between metabolite abundance and position of each sample on a compositional biplot built from SEED subsystem 4 aggregated data were calculated using the kendall method. These correlations were then scaled according to the first and second component of this biplot and plotted according to their scaled position. A. Metabolites detected by GC-MS. B. Metabolites detected by NMR. Only metabolites which differed significantly between BV and health are shown. GHB: gamma-hydroxybutyrate, 2HG: 2-hydroxyglutarate, 5AV: 5-aminovalerate, 2HIC: 2-hydroxyisocaproate, Tyr: tyramine, Cad: cadaverine, Pip: pipecolate, Gluc: glucose, Malt: maltose, Fruc: fructose, TMA: trimethylamine, Lact: lactate, Acet: acetate, Form: formate, 2HV: 2-hydroxyisovalerate, Chol: choline, Succ: succinate.}
\label{F5:metabolite_biplot}
\end{figure}


\begin{table}[!ht]
\begin{adjustwidth}{-2.25in}{0in} % Comment out/remove adjustwidth environment if table fits in text column.
\caption{
{\bf Table caption Nulla mi mi, venenatis sed ipsum varius, volutpat euismod diam.}}
\begin{tabular}{|l|l|l|l|l|l|l|l|}
\hline
\multicolumn{4}{|l|}{\bf Heading1} & \multicolumn{4}{|l|}{\bf Heading2}\\ \hline
$cell1 row1$ & cell2 row 1 & cell3 row 1 & cell4 row 1 & cell5 row 1 & cell6 row 1 & cell7 row 1 & cell8 row 1\\ \hline
$cell1 row2$ & cell2 row 2 & cell3 row 2 & cell4 row 2 & cell5 row 2 & cell6 row 2 & cell7 row 2 & cell8 row 2\\ \hline
$cell1 row3$ & cell2 row 3 & cell3 row 3 & cell4 row 3 & cell5 row 3 & cell6 row 3 & cell7 row 3 & cell8 row 3\\ \hline
\end{tabular}
\begin{flushleft} Table notes Phasellus venenatis, tortor nec vestibulum mattis, massa tortor interdum felis, nec pellentesque metus tortor nec nisl. Ut ornare mauris tellus, vel dapibus arcu suscipit sed.
\end{flushleft}
\label{table1}
\end{adjustwidth}
\end{table}

\section*{Acknowledgments}
The VOGUE Research Group is Deborah Money, Alan Bocking, Sean Hemmingsen, Janet Hill, Gregor Reid, Tim Dumonceaux, Gregory Gloor, Matthew Links, Kieran O'Doherty, Patrick Tang, Julianne Van Schalkwyk and Mark Yudin. We thank Jennifer Reid, Shinthujah Arulanantham and Yohanna Emun for assistance with data compilation.


Funding Statement
Financial support for this study was provided by a joint Canadian Institutes of Health Research (CIHR) Emerging Team Grant and a Genome British Columbia (GBC) grant awarded to DM, SMH, GR and JEH (grant reference \#108030). The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.

\nolinenumbers

%\section*{References}
% Either type in your references using
% \begin{thebibliography}{}
% \bibitem{}
% Text
% \end{thebibliography}
%
% OR
%
% Compile your BiBTeX database using our plos2015.bst
% style file and paste the contents of your .bbl file
% here.
% %%%
% DON'T TOUCH THIS UNTIL WE'RE DONE
\bibliography{bibdesk_refs}

\begin{thebibliography}{10}
\bibitem{bib1}
Devaraju P, Gulati R, Antony PT, Mithun CB, Negi VS. Susceptibility to SLE in South Indian Tamils may be influenced by genetic selection pressure on TLR2 and TLR9 genes. Mol Immunol. 2014 Nov 22. pii: S0161-5890(14)00313-7. doi: 10.1016/j.molimm.2014.11.005

\bibitem{bib2}
Huynen MMTE, Martens P, Hilderlink HBM. The health impacts of globalisation: a conceptual framework. Global Health. 2005;1: 14. Available: http://www.globalizationandhealth.com/content/1/1/14.

\end{thebibliography}



\end{document}

