% Template for PLoS
% Version 3.1 February 2015
%
% To compile to pdf, run:
% latex plos.template
% bibtex plos.template
% latex plos.template
% latex plos.template
% dvipdf plos.template
%
% % % % % % % % % % % % % % % % % % % % % %
%
% -- IMPORTANT NOTE
%
% This template contains comments intended 
% to minimize problems and delays during our production 
% process. Please follow the template instructions
% whenever possible.
%
% % % % % % % % % % % % % % % % % % % % % % % 
%
% Once your paper is accepted for publication, 
% PLEASE REMOVE ALL TRACKED CHANGES in this file and leave only
% the final text of your manuscript.
%
% There are no restrictions on package use within the LaTeX files except that 
% no packages listed in the template may be deleted.
%
% Please do not include colors or graphics in the text.
%
% Please do not create a heading level below \subsection. For 3rd level headings, use \paragraph{}.
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% -- FIGURES AND TABLES
%
% Please include tables/figure captions directly after the paragraph where they are first cited in the text.
%
% DO NOT INCLUDE GRAPHICS IN YOUR MANUSCRIPT
% - Figures should be uploaded separately from your manuscript file. 
% - Figures generated using LaTeX should be extracted and removed from the PDF before submission. 
% - Figures containing multiple panels/subfigures must be combined into one image file before submission.
% For figure citations, please use "Fig." instead of "Figure".
% See http://www.plosone.org/static/figureGuidelines for PLOS figure guidelines.
%
% Tables should be cell-based and may not contain:
% - tabs/spacing/line breaks within cells to alter layout or alignment
% - vertically-merged cells (no tabular environments within tabular environments, do not use \multirow)
% - colors, shading, or graphic objects
% See http://www.plosone.org/static/figureGuidelines#tables for table guidelines.
%
% For tables that exceed the width of the text column, use the adjustwidth environment as illustrated in the example table in text below.
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% -- EQUATIONS, MATH SYMBOLS, SUBSCRIPTS, AND SUPERSCRIPTS
%
% IMPORTANT
% Below are a few tips to help format your equations and other special characters according to our specifications. For more tips to help reduce the possibility of formatting errors during conversion, please see our LaTeX guidelines at http://www.plosone.org/static/latexGuidelines
%
% Please be sure to include all portions of an equation in the math environment.
%
% Do not include text that is not math in the math environment. For example, CO2 will be CO\textsubscript{2}.
%
% Please add line breaks to long display equations when possible in order to fit size of the column. 
%
% For inline equations, please do not include punctuation (commas, etc) within the math environment unless this is part of the equation.
%
% % % % % % % % % % % % % % % % % % % % % % % % 
%
% Please contact latex@plos.org with any questions.
%
% % % % % % % % % % % % % % % % % % % % % % % %

\documentclass[10pt,letterpaper]{article}
\usepackage[top=0.85in,left=2.75in,footskip=0.75in]{geometry}

% Use adjustwidth environment to exceed column width (see example table in text)
\usepackage{changepage}

% Use Unicode characters when possible
\usepackage[utf8]{inputenc}

% textcomp package and marvosym package for additional characters
\usepackage{textcomp,marvosym}

% fixltx2e package for \textsubscript
\usepackage{fixltx2e}

% amsmath and amssymb packages, useful for mathematical formulas and symbols
\usepackage{amsmath,amssymb}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

% Use nameref to cite supporting information files (see Supporting Information section for more info)
\usepackage{nameref,hyperref}

% line numbers
\usepackage[right]{lineno}

% ligatures disabled
\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% rotating package for sideways tables
\usepackage{rotating}

% Remove comment for double spacing
%\usepackage{setspace} 
%\doublespacing

% Text layout
\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.25in 
\textheight 8.75in

% Bold the 'Figure #' in the caption and separate it from the title/caption with a period
% Captions will be left justified
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off]{caption}

% Use the PLoS provided BiBTeX style
\bibliographystyle{plos2015}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

% Leave date blank
\date{}

% Header and Footer with logo
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
\lhead{\includegraphics[width=2.0in]{PLOS-submission.eps}}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}
\lfoot{\sf PLOS}

%% Include all macros below

\newcommand{\lorem}{{\bf LOREM}}
\newcommand{\ipsum}{{\bf IPSUM}}

%% END MACROS SECTION


\begin{document}
\vspace*{0.35in}

% Title must be 250 characters or less.
% Please capitalize all terms in the title except conjunctions, prepositions, and articles.
\begin{flushleft}
{\Large
\textbf\newline{Sumthin' cute here}
}
\newline
% Insert author names, affiliations and corresponding author email (do not include titles, positions, or degrees).
\\
Jean M. Macklaim\textsuperscript{1,},
Amy McMillan\textsuperscript{2,},
Name3 Surname\textsuperscript{2,\textcurrency a},
Name4 Surname\textsuperscript{2,\ddag},
Name5 Surname\textsuperscript{2,\ddag},
Gregor Reid\textsuperscript{2},
Gregory B. Gloor\textsuperscript{3,*},
with the Lorem Ipsum Consortium\textsuperscript{\textpilcrow}
\\
\bigskip
\bf{1} Affiliation Dept/Program/Center, Institution Name, City, State, Country
\\
\bf{2} Affiliation Dept/Program/Center, Institution Name, City, State, Country
\\
\bf{3} Affiliation Dept/Program/Center, Institution Name, City, State, Country
\\
\bigskip

% Insert additional author notes using the symbols described below. Insert symbol callouts after author names as necessary.
% 
% Remove or comment out the author notes below if they aren't used.
%
% Primary Equal Contribution Note
%\Yinyang These authors contributed equally to this work.

% Additional Equal Contribution Note
% Also use this double-dagger symbol for special authorship notes, such as senior authorship.
\ddag Corresponding Author

% Current address notes
\textcurrency a Insert current address of first author with an address update
% \textcurrency b Insert current address of second author with an address update
% \textcurrency c Insert current address of third author with an address update

% Deceased author note
%\dag Deceased

% Group/Consortium Author Note
\textpilcrow Membership list can be found in the Acknowledgments section.

% Use the asterisk to denote corresponding authorship and provide email address in note below.
* CorrespondingAuthor@institute.edu

\end{flushleft}
% Please keep the abstract below 300 words
\section*{Abstract}
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur eget porta erat. Morbi consectetur est vel gravida pretium. Suspendisse ut dui eu ante cursus gravida non sed sem. Nullam sapien tellus, commodo id velit id, eleifend volutpat quam. Phasellus mauris velit, dapibus finibus elementum vel, pulvinar non tellus. Nunc pellentesque pretium diam, quis maximus dolor faucibus id. Nunc convallis sodales ante, ut ullamcorper est egestas vitae. Nam sit amet enim ultrices, ultrices elit pulvinar, volutpat risus.


% Please keep the Author Summary between 150 and 200 words
% Use first person. PLOS ONE authors please skip this step. 
% Author Summary not valid for PLOS ONE submissions.   
%\section*{Author Summary}
%The meta-transcriptome of twenty-two human vaginal samples was characterized at 
%\linenumbers

\section*{Introduction}
The vaginal microbiome is important

The vaginal microbiome has been characterized using 16S rRNA gene sequencing with the result that ...

The vaginal meta-transcriptome was characterized using a very small sample size ...

The vaginal metabolome has been characterized with the result that (confirms and extends) ...

Here we report the integrated analysis of the vaginal microbiome, meta-transcriptome and metabolome on an overlapping set of samples from (N) women from London Ontario. We find that the ..., and .... We suggest that ...

%\begin{equation}\label{eq:schemeP} 
%D_{coll} = \frac{D_f+\frac{[S]^2}{K_D S_T} D_S} {1+\frac{[S]^2}{K_D S_T}}, 
%D_{sm} = \frac{D_f+ \frac{[S]}{K_D} D_S}{1+\frac{[S]}{K_D}},
%\end{equation}

% You may title this section "Methods" or "Models". 
% "Models" is not a valid title for PLoS ONE authors. However, PLoS ONE
% authors may use "Analysis" 
\section*{Materials and Methods}
\subsection*{Ethics.}
\subsection*{Collection.}
\subsection*{Metabolome determination.}
\subsection*{RNA isolation and sequencing.}
\subsection*{Data analysis.} 

\paragraph{Read Mapping.} XX genomes comprising the set of species observed in the human vagina from 16S rRNA gene sequencing (REFS), culture studies (REFS) and additional genomes from organisms suspected to occur were downloaded from Genbank on ????. Open reading frames from these genomes were clustered using ?? with the following criteria (length, PID, etc) and a representative sequence was chosen to be the centroid using the following rule: ???. Centroid sequences were annotated by BLAST to SEED and KEGG databases and the best hit supplying the annotation. The taxonomy of the centroid sequences was taken to represent the taxonomy of the cluster. We will refer to the centroid sequence as the `transcript'. Supplementary Table S1 contains the list of accession numbers. Supplementary Table S2 contains the set of centroid sequences and their inferred annotation. 

Reads in fastq format were mapped to the library of centroid sequences using bowtie2 \cite{bowtie2} and unmapped reads were assembled with ???. The assembled fragments were annotated as above, and added to the master table of counts per sequence feature in each sample. In the end there were an average of XXX million reads mapped to each sample, and Table S3 contains the pertinent information. Reads were further grouped by KEGG function or SEED level 4 subsystem to produce two tables of functions, these will be referred to as `functions'.

\paragraph{Statistical Model for RNA-seq.} The table of counts mapped to reference sequences, and aggregated to either the SEED subsystem or KEGG KO level, appears on the surface to be a table of counts. Common practice is to normalize the  reads per feature such that the `sequencing effort' is a constant or each sample \cite{Anders:2013aa}. However, we  have found this approach  fails in meta-RNA sequencing because of the interplay between the organism and transcript abundance \cite{fernandes:2013,fernandes:2014,Lovell:2015}. Thus, it much more informative to model the observed count for a feature in a sample as a distribution of probabilities that the count was observed given the total number of sequence reads obtained per sample \cite{fernandes:2013}. Such a model falls into the count compositional data analysis paradigm \cite{Aitchison:1986} where only the relative differences between abundances in a sample are informative \cite{pawlowsky2015modeling,pawlowsky2011compositional}. Upon reflection, this approach makes intrinsic sense because the number of reads obtained in any high throughput sequencing experiment is constrained by the capacity of the instrument. We know intuitively that the same library will give a more accurate depiction of reality with the $\sim 200$ million reads obtained from a HiSeq lane than with the $\sim 20$ million reads obtained from a MiSeq.

Compositional data has substantially different properties than do unconstrained data. The most important being that compositional data exhibits `spurious correlation' between features because the possible values for  the features are constrained by the constant sum \cite{pawlowsky2015modeling,pawlowsky2011compositional}. Correlations are also unreliable because they change in unpredictable ways when a subset of the parts are examined. Thus, traditional correlation metrics are unsuitable when examining these data \cite{Lovell:2015}. Note that  all RNA-seq datasets are subcompositions unless ribosomal RNA, tRNA, etc are not depleted and are included in the analysis \cite{pawlowsky2015modeling,pawlowsky2011compositional}. 

Some of these problems can be ameliorated by log-ratio transformations such as the centered log-ratio transformation (clr). Given a vector of numbers that contains D features, $x=[x_1,x_2, ... x_D]$,   $x$ can be converted to a vector of clr values, $z$, as follows: 
\begin{equation}\label{eq:clr}
\mathrm{clr}(x)=\left(\ln \frac{x_1}{g_x}, \ln\frac{x_2}{g_x}, \dots, \ln\frac{x_D}{g_x}\right)=\left(z_1,z_2, \dots, z_n \right) \,.
\end{equation}

Where $g_x$ is the geometric mean of vector $x$.

One complication is that the geometric mean cannot be computed when the $x$ contains a value of 0. Thus two approaches were taken. First, when a point estimate was required  the `count zero multiplicative' zero replacement method from the zCompositions R package \cite{PalareaAlbaladejo201585} was used to adjust 0 values for the likelihood that the 0 represents a non-detect event in the sample. Second, when conducting quantitative analyses, the joint probability distributions for all features in a sample were generated by Dirichlet multinomial sampling within the ALDEx2 Bioconductor R package  \cite{fernandes:2013} using a uniform adjustment of Jeffrey's Prior. This returns a distribution of possible values for the probability of  the feature given the feature count and the total count for the sample \cite{fernandes:2013, gloorAJS:2016}. We have shown previously that the relative error in these probability distributions is largest for features with 0 counts and smallest for those with the largest number of counts in both real and simulated data \cite{fernandes:2013}. The distributions were subsequently transformed using the centre log-ratio transformation  prior to further analysis. 

Exploratory data analysis was performed as point estimates with compositional biplots \cite{aitchison2002biplots} following clr transformation of the data. These show both the distances between samples and the variances of the transcripts.

The expected value of $\phi$ \cite{Lovell:2015} from the clr distribution was used to determine compositionally-linked transcripts or functions, since it has been shown that traditional correlation metrics give unpredictable results\cite{pawlowsky2015modeling,pawlowsky2011compositional}. The expected value of Kendall's Tau (b) was used when reporting the correlation between transcript abundance and metabolite abundance since it is not expected that the metabolome and meta-transcriptome tables share even passingly similar units or scaling. We used an effect size cutoff of 2 when evaluating differential abundance between conditions, a cutoff that was used in our previous investigations \cite{macklaim:2013}. 


\paragraph{Reproducibility.} All R code needed to reproduce this analysis from the count tables can be accessed at: github...


% Results and Discussion can be combined.
\section*{Results and Discussion}

High throughput sequencing experiments generate datasets where the total number of reads per sample are irrelevant, thus these data are compositional and contain only relative information about abundances \cite{fernandes:2014}. Such data can be examined in a rigorous manner by examining the variation in ratios between all pairs of transcripts \cite{Aitchison:1986, pawlowsky2015modeling,pawlowsky2011compositional}. However, this can be dramatically simplified using the centre log-ratio transformation, which reduces each count value to a ratio of that count to the geometric mean count within each sample. The logarithm of the ratio makes any change in relative abundance symmetrical. The clr transformation is functionally equivalent to a matrix of all vs all ratios, and  compensates for differences in read abundance, thus eliminates the need for count normalization \cite{Lovell:2015}.  We and others have shown that clr-transforming the data constitutes the first step towards a compositional data analysis approach that is generally useful  to characterize HTS \cite{Friedman:2012, fernandes:2013,fernandes:2014,Lovell:2015}. All data analysis is thus done in a compositional analysis framework with clr-transformed data as outlined in the Materials and Methods. 

%Two caveats need to be expressed at the outset. First, the clr transformation cannot be performed in the presence of transcripts with 0 counts since it involves taking a logarithm. Initially all transcripts that have 0 reads in all samples, so-called `essential zeros' are removed.  We take two approaches for the remaining 0 value. When a point estimate of 0 is required, we use the count zero multiplicative method \cite{PalareaAlbaladejo201585} that replaces each 0 value using a Bayesian estimate of the most likely value of 0 in each sample. This approach is used to adjust the data for the Singular Value Decomposition that is required for Principle Component Analysis (PCA). When a more robust estimate is required, we generate a Bayesian estimate of the range of values for each transcript (including 0 count transcripts) that are consistent with the original dataset \cite{fernandes:2013}. All subsequent analyses are conducted on the resulting distribution of values, and the final reported value is the expected value of the p-value, $\phi$ \cite{Lovell:2015}, or effect-size metric. 

\subsection{Exploratory analysis}
We collected NN samples for RNA-seq and metabolomic analysis, and MM were sequenced on the Illumina HiSeq platform at The Centre for Applied Genomics in Toronto. We also included in the RNA-seq results four samples sequenced using the ABI SoLiD platform and reported previously \cite{macklaim:2013}. We examined these samples using taxonomic abundance profiles determined both from 16S rRNA gene sequencing and from the reference sequence abundance; using  compositional biplots and unsupervised clustering on clr-transformed data; and by correlating the samples with the clinical phenotype. In the end, we excluded six samples  (Supplemental Text) since they had either a very distinctive taxonomic abundance profile or had a microbiome profile did not match the clinical phenotype. This left 22 samples for RNA-seq analysis composed of three samples sequenced by ABI-SoLid and 19 sequenced by Illumina HiSeq. The table of counts for the twenty-two samples were filtered to remove all transcripts with an average of 2 or fewer reads across all 22 samples, this simplified the dataset from 48000 transcripts to 10052 transcripts. 

Figure \ref{F1:marplot} shows ...

A compositional biplot of this dataset is shown in Fig.~\ref{F2:refseq_biplot}A. Compositional biplots summarize in one plot both the relationships between the samples and the contributions of the variables to those relationships \cite{aitchison2002biplots}. In this plot, the distance of a transcript from the origin is  related to the standard deviation of the clr value of the transcript in the samples: it is \emph{not} directly related to the absolute abundance of the transcript. The direction of the transcript relative to the samples shows the sample group with the greatest abundance of the transcript. Our ability to interpret the distance and direction information is only as good as the projection of the PCR, which is determined by the proportion of variance explained on the plot. The biplot in Figure \ref{F2:refseq_biplot} explains 47.7\% of the variance on PC1 and 11.2\% of the variance on PC2. We conclude that this representation is a good one since these two principle components explain nearly 60\% of the variance in such a large dataset.

There are several features of the data that can be gleaned. First, we can see that the samples partition nicely into two groups, healthy on the left and BV on the right.   This separation is driven by several taxa known to be associated with health (\emph{Lactobacillus} species) on the left and of BV furthest to the right (non-lactobacillus anaerobes).  Interestingly, transcripts annotated as belonging to \emph{Lactobacillus iners} are near the middle of PC1, indicating that transcripts associated with this organism contribute little to the health-BV separation.  Within BV, we observe that \emph{Megasphaera} and \emph{Prevotella} species form two distinct foci suggesting the presence of distinctive species or strains of these organisms in BV. Interestingly, for \emph{Megasphaera} one of these foci is very close to the origin on PC1, suggesting that this strain or species is contributing little to the overall BV phenotype. Finally, several de-novo assembled transcripts appear to be major contributors to the BV phenotype.

We next observe that there are potential groups of healthy and of BV samples that separate along PC2. The first healthy group, H1, is composed of samples 30S, 4S, 1B, 9B, and 10B. The second healthy group, H2, is composed of samples 2B, 4B, 0B, 1A and 6B. The first BV group, BV1, is composed of samples 16B, 27S, 14B, 13A, 8A, 17B, 18B, 12B, and the second, BV2, is composed of samples 9A, 6A, 10A and 12A. The reason for these groupings is obvious when we examine the taxonomic origin of each transcript.  Here we must recall that there is \emph{no information} regarding absolute abundance for a transcript and that two transcripts with widely varying absolute abundances but similar ratio abundances will have similar locations on the biplot. Inspection of the location of the transcripts in the biplot indicates that they are largely separable by taxonomy. Transcripts mapping to \emph{Lactobacillus crispatus}  furthest to the left and near the bottom correspond to the H1 group. Transcripts derived to the remaining lactobacilli are more closely associated with the H2 group. Such a split has been observed before, and group H1 corresponds to vaginal community state type 1 \cite{Ravel:2010}. The samples in H2 are much less homogenous than those in H1, and correspond  to a mixture of community state types two through 4. 

The BV1 group is associated with an increased ratio abundance of \emph{Leptotrichia} sp., one  \emph{Megasphaera} and one \emph{Prevotella} species and the unmapped assembled transcripts. The BV2 group is more associated with \emph{L. iners}, \emph{Gardnerella vaginalis} and the other strain or species of \emph{Prevotella} sp. and \emph{Megasphaera} sp. 

We note that the major lactobacillus groups separate much more strongly on PC2 than do the taxa associated with BV. This could indicate that healthy microbiotas colonized by a near monoculture of one or the other lactobacillus species have distinct ways of being healthy, or it could be an artefact of the non-overlapping gene content of these organisms. The partitioning of different lactobacillus dominated healthy microbiota types is well documented in the literature \cite{Ravel:2010, Hummelin:2010,macmillan:2015}   and we did not examine this further. However, it is rarely observed that the microbiota of the BV condition partitions in a meaningful way, and we examine the causes of this in a later section.

%%%%
% this is figure of biplots_filter.pdf
%%%%
\begin{figure}[h]
\caption{{\bf Exploratory analysis of the reference sequence transcripts.}
Compositional biplots of the reference sequence transcripts filtered to include transcripts present at an average count of more than 2 reads per sample. Compositional biplots are Principle Component (PC) Plots generated from the singular value decomposition of the centre log-ratio transformed dataset \cite{aitchison2002biplots}. The bottom and left axes show the unit scaled measures for PC1 and 2, and the top and right axes show unit scaled variances for the transcripts. These plots thus show the relationship between sample distance along with the contribution of the transcripts to that distance. In this plot component 1 and 2 explain over 58\% of the variance in the dataset, which is exceptional for a dataset of this complexity. Sample names are shown in black text. The location of each transcript is shown as a coloured point, with the colors corresponding to the taxonomic assignment of the centroid sequence. The second panel shows a blow-up of the area of the biplot on the right side. Taxa are labeled with the  color corresponding to the points and the following legend: Lc-\emph{L. crispatus},  Li-\emph{L. iners},  Lje-\emph{L. jensenii}, Ljg-\emph{L. johnsonii} or \emph{L. gasseri}, Gv-\emph{G. vaginalis}, Me-\emph{Megasphaera} sp., Pr-\emph{Prevotella} sp., AS- assembled transcripts. }
\label{F2:refseq_biplot}
\end{figure}

Not surprisingly, the reference sequence-based biplot shows that taxonomic abundance is the major driver of the variation of transcript abundance between samples. We were interested to determine if the different states had different underlying functions regardless of the taxonomic composition. Thus the reads were grouped by SEED subsystem 4 function or by KEGG function and Fig. \ref{F3:function_biplot} shows the result.  Here we can see that the functions partition more strongly along the major axis of variance with 56.2\% of the variance explained on PC1, and 8.5\% on PC2 when aggregated by SEED subsystem and 57.3\% and 8.6\% when aggregated by KEGG function. Thus the first two components explain at least 64\% of the variance regardless of the functional classification used. We observe that in both cases the  samples group into the same four groups as observed for the non-aggregated data. This indicates that the observed splits are robust to aggregation of the data, and are not simply driven by changes in taxonomic abundance. Samples composing the healthy group on the left side of the biplot are associated with a relatively small set of functions that are proportionally increased in these samples, and the samples composing the BV group are associated with a large set of functions that are proportionally increased in the BV samples.  The large number of functions with small variance indicates the presence of a core set of functions that are required regardless of condition. The health and BV samples also partition along PC2 similarly to the partitioning seen in Panel A. 

We note that that the majority of functions are at or near the origin with the greatest density being near -5 on PC1. A limitation of the a centered log-ratio approach, or indeed of any ratio-based approach is that the geometric mean of a sample depends on the density of 0 values in the sample.  Samples will have a high density of 0 values in the group that has fewer expressed genes. The H1 and H2 group samples, composed of \emph{Lactobacillus} species will have a much smaller set of functions than will the BV1 and BV2 groups that comprise a mixed bag of organisms, and which generally include at least some functions found only in \emph{Lactobacillus}. Thus the geometric mean for the H groups will be closer to 0 than will the geometric mean for the BV group samples, and the values will have higher . The clr value for the  function i in H and BV, is determined by the formula $\frac{F_i}{gm}$, and so genes with low counts will \emph{appear} to be exhibit variation indicating a greater abundance on the H side of PC1.

\subsection{Exploring differential abundance}

Traditional statistical approaches, while widely used, are inappropriate when examining compositional  abundances using the $\phi$ metric that assesses the stability of the abundance ratios across samples for all possible pairs of transcripts \cite{Lovell:2015}. Samples that have stable ratios are said to be compositionally associated and have a value of $phi$ near 0. 

As with any compositional data analysis approach, values of 0 are not compatible with the transformation requried. In the original paper, the authors removed from analysis all genes that had a 0 count in any sample and then calculated the clr and $\phi$ as point estimates. We took a different approach and estimated the distribution of values that 0 could reasonably assume and then clr-transformed the data using the aldex.clr function from the ALDEx2 Bioconductor package \cite{fernandes:2013, fernandes:2014}. The $\phi$ measure was then calculated for each independent element of the distribution and the expected value of $\phi$ was reported. This approach allowed us to determine reasonable estimates for compositional association values even for functions with a dichotomous distribution, while still maintaining a low false positive background. 

Figure \ref{F3:phi_biplot} shows a graphical depiction of $\phi$ groupings determined for the SEED subsystem 4 functional grouping and the KEGG functional grouping. In both approaches, there were four major clusters that partitioned on PC1, and Supplementary Tables 3 and 4 contain the clusters of SEED subsystem 4 and KO annotations. The first group, composed of XX members was the small group of functions that were much more relatively abundant in the healthy than in the BV group. Inspection of these functions shows that they are belong to ???. These same functions were generally observed to be differentially abundant in our previous analysis (REF). The second group which is nearer the origin was also relatively more abundant in health than in BV, and were found to correspond largely to replication and gene expression housekeeping genes. This confirms our previous observation that these abundant housekeeping genes are slightly increased relative tot he remainder in the healthy microbiome. The third major group had a much greater relative abundance in BV than in health. These functions are largely functions that are absent in lactobacilli and correspond to functions involved in the greater biosynthetic and respiratory capacities of the majority of the BV-associated organisms (REF). In addition to the three major groups, there were a number of smaller clusters. These were generally found to correspond to functions in the same pathway or in many cases to co-expressed subunits.  A full list of functions and their associated group is found in tabular format in the supplement. 


\begin{figure}[h]
\caption{{\bf Compositional association of microbial functions.}
Panel A shows a compositional biplot of the SEED subsystem 4 aggregated data with different groups of compositionally associated transcripts coloured by group membership. Panel B shows the same for data aggregated and analyzed by KEGG annotation. In both cases we determined the expected value of $\phi$ with a cutoff of 0.03. }
\label{F3:phi_biplot}
\end{figure}




\begin{table}[!ht]
\begin{adjustwidth}{-2.25in}{0in} % Comment out/remove adjustwidth environment if table fits in text column.
\caption{
{\bf Table caption Nulla mi mi, venenatis sed ipsum varius, volutpat euismod diam.}}
\begin{tabular}{|l|l|l|l|l|l|l|l|}
\hline
\multicolumn{4}{|l|}{\bf Heading1} & \multicolumn{4}{|l|}{\bf Heading2}\\ \hline
$cell1 row1$ & cell2 row 1 & cell3 row 1 & cell4 row 1 & cell5 row 1 & cell6 row 1 & cell7 row 1 & cell8 row 1\\ \hline
$cell1 row2$ & cell2 row 2 & cell3 row 2 & cell4 row 2 & cell5 row 2 & cell6 row 2 & cell7 row 2 & cell8 row 2\\ \hline
$cell1 row3$ & cell2 row 3 & cell3 row 3 & cell4 row 3 & cell5 row 3 & cell6 row 3 & cell7 row 3 & cell8 row 3\\ \hline
\end{tabular}
\begin{flushleft} Table notes Phasellus venenatis, tortor nec vestibulum mattis, massa tortor interdum felis, nec pellentesque metus tortor nec nisl. Ut ornare mauris tellus, vel dapibus arcu suscipit sed.
\end{flushleft}
\label{table1}
\end{adjustwidth}
\end{table}




\section*{Supporting Information}

% Include only the SI item label in the subsection heading. Use the \nameref{label} command to cite SI items in the text.
\subsection*{S1 Video}
\label{S1_Video}
{\bf Bold the first sentence.}  Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

\subsection*{S1 Text}
\label{S1_Text}
{\bf Lorem Ipsum.} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

\subsection*{S1 Fig}
\label{S1_Fig}
{\bf Lorem Ipsum.} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

\subsection*{S2 Fig}
\label{S2_Fig}
{\bf Lorem Ipsum.} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

\subsection*{S1 Table}
\label{S1_Table}
{\bf Lorem Ipsum.} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

\section*{Acknowledgments}
Cras egestas velit mauris, eu mollis turpis pellentesque sit amet. Interdum et malesuada fames ac ante ipsum primis in faucibus. Nam id pretium nisi. Sed ac quam id nisi malesuada congue. Sed interdum aliquet augue, at pellentesque quam rhoncus vitae.

\nolinenumbers

%\section*{References}
% Either type in your references using
% \begin{thebibliography}{}
% \bibitem{}
% Text
% \end{thebibliography}
%
% OR
%
% Compile your BiBTeX database using our plos2015.bst
% style file and paste the contents of your .bbl file
% here.
% %%%
% DON'T TOUCH THIS UNTIL WE'RE DONE
\bibliography{bibdesk_refs}

\begin{thebibliography}{10}
\bibitem{bib1}
Devaraju P, Gulati R, Antony PT, Mithun CB, Negi VS. Susceptibility to SLE in South Indian Tamils may be influenced by genetic selection pressure on TLR2 and TLR9 genes. Mol Immunol. 2014 Nov 22. pii: S0161-5890(14)00313-7. doi: 10.1016/j.molimm.2014.11.005

\bibitem{bib2}
Huynen MMTE, Martens P, Hilderlink HBM. The health impacts of globalisation: a conceptual framework. Global Health. 2005;1: 14. Available: http://www.globalizationandhealth.com/content/1/1/14.

\end{thebibliography}



\end{document}

